src/intellisense/diagnosticsProvider.ts

```diff
@@ -66,7 +67,9 @@ export class NotebookCellBangInstallDiagnosticsProvider
         this.disposables.push(languages.registerHoverProvider(PYTHON_LANGUAGE, this));
         this.documents.onDidChangeTextDocument(
             (e) => {
-                const notebook = e.document.notebook;
+                const notebook = workspace.notebookDocuments.find((notebook) =>
+                    notebook.getCells().some((cell) => cell.document === e.document)
```

Comment:
I feel like if 'document.notebook' has gone away, this might ripple a lot further than this?

Reply:
Damn it, that's supposed to go into a separate PR,